src/scripts/calibration_analyses/analysis_scripts/analysis_demography_calibrations.py

```diff
@@ -294,7 +294,7 @@ def get_mean_pop_by_age_for_sex_and_year(sex, year):
             fig.show()
             plt.close(fig)
 
-        except pd.core.base.DataError:
+        except pd.errors.DataError:
```

Comment:
I also need help to understand the intent of this `try`-`except` block: was it to catch cases where the dataframe was empty, because `year` was after the simulation's end?  Because now I think we're getting an empty dataframe in that case, instead of a `DataError` exception being raised, so that this entire `try`-`except` block might be completely useless, and things should be handled differently

Reply:
That's right --- the simulations that this script is run for may be of different lengths, and so may or may not include some of  the years `[2010, 2015, 2018, 2029, 2049]`, for which this routine is called.

I suppose a better check would be of the form:

`if year in wpp_ann['Year']: `